drivers/vmwarevsphere/vsphere.go

```diff
@@ -321,6 +328,24 @@ func (d *Driver) PreCreateCheck() error {
 
 	f.SetDatacenter(dc)
 
+	// Folder
+	if d.Folder != "" {
+		// Find the specified Folder to create the VM in.
+		folders, err := dc.Folders(ctx)
+		if err != nil {
+			return err
+		}
+		f, err := object.NewSearchIndex(c.Client).FindChild(ctx, folders.VmFolder, d.Folder)
```

Comment:
This line is why @cwhits can't use a sub-directory as the destination folder for the VM. As a TODO for me this is what needs fixing (possibly needs to [Folder](https://godoc.org/github.com/vmware/govmomi/find#Finder.Folder)